.github/workflows/publish-branch.yaml

```diff
@@ -0,0 +1,77 @@
+
+on:
+    push:
+        branches:
+            - main
+            - dev
+
+jobs:
+    release_candidate:
+        runs-on: ubuntu-latest
+        env:
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+            NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.head_ref }}
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: "16.x"
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "yarn"
+
+            - name: Install dev dependencies
+              run: yarn install
+
+            - name: Setup npm credentials file
+              run: echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" >> .npmrc
+
+            # - name: Setup git credentials
+            #   run: |
+            #       git config --global user.name 'Auto Release Bot'
+            #       git config --global user.email 'auto-release-bot@users.noreply.github.com'
+
+            - name: Get current package.json version
+              run: echo "PACKAGE_VERSION=$(npm pkg get version | tr -d '"')" >> $GITHUB_ENV
+
+            # get the sort SHA and add it into the environment variables
+            - name: Get short SHA
+              run: echo "SHORT_SHA=${GITHUB_SHA::7}" >> $GITHUB_ENV
+
+            # get the package name so the workflow can be reusable between projects
+            # - name: Get current package.json name
+            #   run: echo "PACKAGE_NAME=$(npm pkg get name | tr -d '"')" >> $GITHUB_ENV
+
+            - name: Setup Branch Release Candidate Version
+              run: echo "NEW_VERSION=$PACKAGE_VERSION-$BRANCH_NAME.$SHORT_SHA" >> $GITHUB_ENV
+
+            - name: Update Package.json Version with Release Candidate Version
+              run: node ./update-branch-version.js --version $NEW_VERSION
+
+            - name: Publish a new branch release candidate version
+              run: npm publish --tag $BRANCH_NAME
+              env:
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+            # This comment only works for Pull Requests
+            # - uses: actions/github-script@v6
+            #   with:
+            #     script: |
+            #       github.rest.issues.createComment({
+            #         issue_number: context.issue.number,
+            #         owner: context.repo.owner,
+            #         repo: context.repo.repo,
+            #         body: `Release candidate released with the last commit 🚀
+
+            #         \`\`\`
+            #         yarn add ${{ env.PACKAGE_NAME}}@${{ env.NEW_VERSION }}
+            #         \`\`\`
+            #         or
+            #         \`\`\`
+            #         npm install ${{ env.PACKAGE_NAME}}@${{ env.NEW_VERSION }}
+            #         \`\`\`
+            #         `
+            #       })
```

Comment:
The commented code on this workflow can be used to add a comment on some discussion or PR (currently is implemented for PRs), but please feel free to remove it if not needed

Reply:
Yeah I think this is okay to remove.